pkg/agent/flowexporter/exporter/exporter_perf_test.go

```diff
@@ -160,7 +161,12 @@ func setupExporter(isConntrackConn bool) (*FlowExporter, error) {
 		IdleFlowTimeout:        testIdleFlowTimeout,
 		StaleConnectionTimeout: 1,
 		PollInterval:           1}
-	exp, _ := NewFlowExporter(nil, nil, nil, nil, config.TrafficEncapModeEncap, nil, nil, nil, nil, false, nil, o)
+	_, podIPv4CIDR, _ := net.ParseCIDR("100.100.100.0/24")
+	nodeConfig := &config.NodeConfig{
+		PodIPv4CIDR: podIPv4CIDR,
+	}
+	ovsDatapathType := ovsconfig.OVSDatapathNetdev
```

Comment:
I'm not sure about if it's ok to set `ovsDatapathType` to `ovsconfig.OVSDatapathNetdev`. Will it fail on different OS setup? Is it better for me to add a nil check in [conntrack.go](https://github.com/antrea-io/antrea/blob/e86d5e9637477a2a924c1e6eeb8a4b6d758d563d/pkg/agent/flowexporter/connections/conntrack.go#L28) so we can pass nil as the `ovsDatapathType` here?

Reply:
what was the behavior of the test before the initialization changes? what kind of connection store were we using?

BTW, it's strange that ovsDatapathType is a pointer in `NewFlowExporter`. I think we should pass the value directly.

Reply:
previously the connection store initialization is separated from flowexporter, and connTrackDumper is an explicit parameter of conntrackConnStore initialization. The previous test initialized a `conntrackConnStore` without a connTrackDumper, as we are adding conns manually into the connection map, we don't actually need a connTrackDumper

I can change it from pointer to value. I just don't know how to properly initialize `ovsDatapathType` here. `OVSDatapathSystem` failed on vagrant cluster:
```
E1105 19:50:32.932909   38568 sysctl_linux.go:64] Error when setting netfilter/nf_conntrack_acct: open /proc/sys/net/netfilter/nf_conntrack_acct: permission denied
E1105 19:50:32.933355   38568 sysctl_linux.go:64] Error when setting netfilter/nf_conntrack_timestamp: open /proc/sys/net/netfilter/nf_conntrack_timestamp: permission denied
```

 `OVSDatapathNetdev` works fine, so I put `OVSDatapathNetdev` here. But I'm not familiar with it.

Reply:
You don't *have* to use `NewFlowExporter` in the benchmark definition. If there is a way to initialize sub-components directly that makes more sense for the benchmark, you should go that route.

Reply:
I don't think currently there's a good way to do that. As we need to initialize sub-components, and we also need to initialize `FlowExporter`. A dumb way is adding at least one more parameter to `NewFlowExporter`, indicating whether the connection stores are initialized outside of the method, if true, then we skip the initialization of them in the method.

Reply:
IMO it would be ok to have a dedicated initialization function for testing that you can tweak as needed:
```
func NewFlowExporter(o *flowexporter.FlowExporterOptions) (*FlowExporter, error) {
	// Initialize IPFIX registry
	registry := ipfix.NewIPFIXRegistry()
	registry.LoadRegistry()

	// Prepare input args for IPFIX exporting process.
	nodeName := "test-node:
	expInput := prepareExporterInputArgs(o.FlowCollectorAddr, o.FlowCollectorProto, nodeName)

	v4Enabled := true
	v6Enabled := false

	denyConnStore := connections.NewDenyConnectionStore(nil, nil, o)
	conntrackConnStore := connections.NewConntrackConnectionStore(nil, v4Enabled, v6Enabled, nil, nil, false, o)

	return &FlowExporter{
		conntrackConnStore:     conntrackConnStore,
		denyConnStore:          denyConnStore,
		registry:               registry,
		v4Enabled:              v4Enabled,
		v6Enabled:              v6Enabled,
		exporterInput:          expInput,
		ipfixSet:               ipfixentities.NewSet(false),
		k8sClient:              nil,
		nodeRouteController:    nil,
		isNetworkPolicyOnly:    false,
		nodeName:               nodeName,
		conntrackPriorityQueue: conntrackConnStore.GetPriorityQueue(),
		denyPriorityQueue:      denyConnStore.GetPriorityQueue(),
		expiredConns:           make([]flowexporter.Connection, 0, maxConnsToExport*2),
	}, nil
}
```

would something like this work?

Reply:
Thanks, Antonin. Yes, having a dedicated function for testing purpose makes sense to me. 

Updated the PR.